### DISCUSSION TITLE
关于androidTV默认优先走http DNS的问题

### DISCUSSION THREAD

**Discussion by bsdcpp** [MetaCubeX/mihomo#1628] [General]
我有台索尼电视（国行应该是androidTV阉割版），之前我给配的dns是223.5.5.5，它会优先走http解析，然后导致mihomo无法劫持到。后来我在路由器上分配运营商自己dns，问题解决。
但是现在换了openwrt，似乎默认dns分配的就是路由器本身地址，可能和dnsmasq默认有关。这样一来问题又回来了，我也没有配置223.5.5.5相关的dns。

现在在clash api上看到的都是ip，解析过的，无法进行域名匹配，开嗅探也没用，请问有什么优雅点的办法避免这样的问题，谢谢。

**Comment by fejeIaqbhbv** [comment]
mihomo在openwrt上？
找不到源头就写个防火墙规则把223.5.5.5给劫了

**Reply by fejeIaqbhbv** [reply]
我前几天配adghome也遇到了类似情况，不过我设备都走的53没走doh
https://github.com/istoreos/istoreos/discussions/1871#discussioncomment-11083755

**Reply by bsdcpp** [reply]
感谢回复，我目前也就是在android tv上遇到这个问题，没有配doh，223默认带http dns，电视会优先去走。
我先在br-lan的DHCP里配了option 6,dns1,dns2，这样分配运营商地址好了

**Reply by fejeIaqbhbv** [reply]
那挺好
我还以为你和我一样是系统内置级的，option 6没有用，无论怎么指定dns，都没法覆盖，只能劫持

**Reply by bsdcpp** [reply]
> 那挺好 我还以为你和我一样是系统内置级的，option 6没有用，无论怎么指定dns，都没法覆盖，只能劫持

对的，除了53，还有http的dns也要劫持，一般工具只对udp 53劫持
